### DISCUSSION TITLE
How do I render an interface entry in extension like Directus does?

### DISCUSSION THREAD

**Discussion by Pachat** [directus/directus#17219] [Archive]
Following the documentation to create an extension for an input field, the SDK builds the interface.vue this way
```
<template>
		<input :value="value" @input="handleChange($event.target.value)" />
</template>

```
![input-rendering-1](https://user-images.githubusercontent.com/13746133/213210276-c1adbc0c-efb1-4b54-ae2b-87c10cc0aca8.png)
As you can see the two inputs, the first being a normal Directus input, the second with the expansion interface are not alike.

If I inject in the template the missing divs taken from the regular Directus input like this
```
<template>
  <!----><div class="v-input full-width sans-serif" data-v-0b6f9ad5="" data-v-9ac7f983="" data-v-c2ed0384=""><!---->
    <div class="input" data-v-0b6f9ad5=""><!----><!---->
      <input :value="value" @input="handleChange($event.target.value)" /><!----><!----><!---->
    </div><!---->
  </div>
</template>

```
rendering becomes
![input-rendering-2](https://user-images.githubusercontent.com/13746133/213212349-5231f08a-4bc3-4f91-b8f2-e5f6c2e6b999.png)
Still not quite the same.

But the most annoying thing is may be that the generated HTML `<input ... />` is not the same.
![input-rendering-3](https://user-images.githubusercontent.com/13746133/213213708-17fcf1e6-e5df-42ff-b86a-fa824e88f615.png)
corresponding to the first image
and
![input-rendering-4](https://user-images.githubusercontent.com/13746133/213215368-f45dbf3a-e18a-4281-83cc-d322eb56fd7f.png)
corresponding to the second image.

So could you please tell me how to render an interface entry in extension like Directus does for regular input?

**Comment by br41nslug** [comment]
https://components.directus.io/?path=/docs/components-vinput--primary

**Reply by Pachat** [reply]
Great! That's certainly what I need.
Where do I put the code
```
(args) => ({
	setup() {
		return { args };
	},
	template: '<v-input v-bind="args" v-on="args" />',
})
```
in the default `display.vue`?
```
<template>
		<input :value="value" @input="handleChange($event.target.value)" />
</template>

<script>
export default {
	props: {
		value: {
			type: String,
			default: null,
		},
	},
	emits: ['input'],
	setup(props, { emit }) {
		return { handleChange };

		function handleChange(value) {
			emit('input', value);
		}
	},
};
</script>

```

**Reply by br41nslug** [reply]
should be just replacing the `<input` with `<v-input` for the very basic funcitonality 😄

**Reply by Pachat** [reply]
OK. Did it
But when I type in the input box, I just see the last char typed. And only the last char is stored.
```
<template>
	<v-input :value="value" @input="handleChange($event.target.value)" />
</template>

<script>
export default {
	props: {
		value: {
			type: String,
			default: null,
		},
	},
	emits: ['input'],
	setup(props, { emit }) {
		return { handleChange };

		function handleChange(value) {
			console.log(value)
			emit('input', value);
		}
	},
};
</script>
```

With `<input ...` instead of `<v-input ...` I see the whole string I'm typing and it is stored correctly.
Except that the appearance of the input box is not the regular Directus one.
So, apart replacing `<input ...` by `<v-input ...` it looks like some necessary modifications in the code are needed.

**Reply by br41nslug** [reply]
The parameters available for the `v-input` are documented on the previous link. You'll probable want to use either `v-model=` for 2-way binding or `:model-value=` with an input event handler.

**Reply by Pachat** [reply]
Sorry, but what I see from the [doc](https://components.directus.io/?path=/docs/components-vinput--primary) is
1°) `Code`
```
(args) => ({
	setup() {
		return { args };
	},
	template: '<v-input v-bind="args" v-on="args" />',
})

```
I have no idea where it should be used.

2°) `Props`: do they have to be set in the index.js?
3°) `Events`
4°) `Slots`.
I am not advanced enough to bring it together without an example of a complete work of an `index.js` and `display.vue` to simply make an input that appears the Directus way.

Can't I just add CSS to the `<input ...` that works fine?

**Reply by br41nslug** [reply]
Thats just the code snippet used to render the input there, you can ignore that if thats not useful to you 😄 
![image](https://user-images.githubusercontent.com/9389634/213259745-5fd7e62d-aea2-4199-9ae9-108e49a44339.png)

> Can't I just add CSS to the <input ... that works fine?

You definitely can, but it can be a lot of work getting to the same style as `v-input`.

A complete example here then instead of bits and pieces:
```vue
<template>
	<v-input :model-value="value" @input="handleChange($event.target.value)" />
</template>

<script>
export default {
	props: {
		value: {
			type: String,
			default: null,
		},
	},
	emits: ['input'],
	setup(props, { emit }) {
		return { handleChange };

		function handleChange(value) {
			emit('input', value);
		}
	},
};
</script>
```

**Reply by Pachat** [reply]
Thank you so much.
I see my mistake:
I was using `:value="value"`  instead of `:model-value="value" `

Thank you for your time. Very appreciated!
